pkg/daemon/ceph/client/filesystem.go

```diff
@@ -134,16 +144,93 @@ func CreateFilesystem(context *clusterd.Context, clusterName, name, metadataPool
 
 	// set the number of active mds instances
 	if activeMDSCount > 1 {
-		args = []string{"fs", "set", name, "max_mds", strconv.Itoa(int(activeMDSCount))}
-		_, err = ExecuteCephCommand(context, clusterName, args)
-		if err != nil {
+		if err = SetNumMDSRanks(context, clusterName, name, activeMDSCount); err != nil {
 			logger.Warningf("failed setting active mds count to %d. %+v", activeMDSCount, err)
 		}
 	}
 
 	return nil
 }
 
+// IsMultiFSEnabled returns true if ROOK_ALLOW_MULTIPLE_FILESYSTEMS is set to "true", allowing
+// Rook to create multiple Ceph filesystems. False if Rook is not allowed to do so.
+// TODO: multi-fs is always available for Mimic+ clusters
+func IsMultiFSEnabled() bool {
+	t := os.Getenv(MultiFsEnv)
+	if t == "true" {
+		return true
+	}
+	return false
+}
+
+// SetNumMDSRanks sets the number of mds ranks (max_mds) for a Ceph filesystem.
+func SetNumMDSRanks(context *clusterd.Context, clusterName, fsName string, activeMDSCount int32) error {
+	// Noted sections 1 and 2 are necessary for reducing max_mds in Luminous. These two sections can
+	// be removed when Rook no longer supports Luminous.
+	//   See more:   [1] http://docs.ceph.com/docs/luminous/cephfs/upgrading/
+	//               [2] https://tracker.ceph.com/issues/23172
+	// TODO: These two sections could be conditionally skipped if this function can be informed that
+	//   the version of Ceph mdses currently running is >=13.0.0
```

Comment:
Would you be able to test this tomorrow @jan--f and get back to us on whether it is necessary or not? 

@travisn, I propose to make an additional commit in this branch that removes the questionably unnecessary reduce-and-wait functionality. If it ends up being unnecessary, the additional commit can be rebased onto the main mds refactor, and there should be no issue with tests failing. If it is necessary, the extra commit can be removed, and there should be little to no work to make the tests pass again. How's that sound?

Reply:
@BlaineEXE yes, sounds good to split that out into another commit and evaluate if it's still needed.

Reply:
This is the CI that was run before I added the extra commit. Everything passed except one of the integration tests with an issue that I think is part of the CI flakiness. https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/PR-2153/27/pipeline

Reply:
I chatted with Jan this morning, and his recommendation was to leave the shrink-to-upgrade functionality in for now. Jan reported that it seems like the mds has "at least the basic ability to distinguish feature map compatibility" but that "all feature are in the incompat category." This suggests that we should shrink before upgrading if the new version has MDS features added, but unless we keep a version->feature_changed map around we are better off shrinking before every upgrade.

This also came with the note that the mons currently have the capability of upgrading with new features and not causing problems, and Jan believes that the mds should get such a feature at some point. So Jan will passively watch the for if/when this will change and update us then.

So for now, I'll remove the extra commit, and I have a few comments I plan to pare down before pushing what I hope will be the final (or at least penultimate) version of this PR.